bus_board.rb

```diff
@@ -0,0 +1,71 @@
+require 'json'
+require 'net/http'
+require 'uri'
+
+def display_bus(bus)
+  time_until_arrival = bus["timeToStation"]/60 == 0 ? "due" : "#{(bus["timeToStation"]/60).to_s} minutes"
+  puts "line number: #{bus["lineName"]} || destination: #{bus["destinationName"]} || time until arrival: #{time_until_arrival}"
+end
+
+def display_buses_by_station_code(bus_code)
+  url = "https://api.tfl.gov.uk/StopPoint/" + bus_code + "/Arrivals"
+  bus_departures = call_api(url)
+  bus_departures = bus_departures.sort_by { |bus| bus["timeToStation"] }
+
+  if bus_departures.empty?
+    puts "No arrivals in the near future"
+    return
+  end
+
+  max_index = [5, bus_departures.size].min
+  (0..max_index-1).each { |i|
+    bus = bus_departures[i]
+    display_bus(bus)
+  }
+end
+
+def call_api(url)
+  result = Net::HTTP.get(URI.parse(url))
+  JSON.parse(result)
+end
```

Comment:
I wouldn't do field to return since sometimes the result is already just what we need - would it still make sense to do the base url and the path separately